stack/deprecating-interfaces.rst

```diff
@@ -25,13 +25,23 @@ Code removal
 ------------
 
 When an interface is deprecated, a ticket should be filed to remove the corresponding deprecated code, usually assigned to the team of the deprecating developer.
-Removal of the deprecated code occurs, at the earliest, immediately after the next major release following the release with the first deprecation release note; at the latest, immediately before the major release following.
-In other words, if deprecation is first noted in release 17.2.3, the code cannot be removed until after 18.0 is released and must be removed before 19.0 is released.
-So the code removal ticket should block the following major release (19.0 in this example).
-In general, no deprecation should be removed before two calendar months have elapsed.
+Removal of the deprecated code occurs, at the earliest, immediately after the next major release, which will be the first one to include the deprecation in its release note, or after eight weeks (whichever is longer).
+Removal should occur before the major release after the one in which it is first deprecated.
+For the purposes of removal, the "release" event is when the first release candidate is tagged, because this is when the release branch may start to diverge from the main branch.
+In other words, if a deprecation is merged to the main branch between the first release candidate tags for v17 and v18, it may only be removed after the first release candidate tag for v18, and should be removed before the first release candidate tag for v19.
+The code removal ticket should by blocked by the next major release (v18 in this example), and block the following one (v19).
 Scheduling of the code removal should be handled like :doc:`any other backlog story </work/project-planning>`, although with a clear deadline (and a clear "do not merge before" point, unlike most stories).
```

Comment:
Maybe add a paragraph pointing out that users following major releases will see the deprecation announced in the release notes as they start to get warnings at runtime, then have until the next major release to update, while users following weekly releases will see the warnings at runtime and have at minimum 8 weeks to update.